dubbo-monitor/dubbo-monitor-default/src/main/java/org/apache/dubbo/monitor/dubbo/MetricsFilter.java

```diff
@@ -123,6 +127,26 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
         }
     }
 
+    private void reportLbMetrics(Result result, Invoker<?> invoker, boolean isProvider) {
```

Comment:
I think the logic of this method should be done in a dedicated thread, which will periodically update the cpu usage if necessary. 
What could be done here is just get the cpu.user data and put it into attachment.
I also think it is no necessary to create CpuUsageGauge, just use com.alibaba.metrics.os.linux.CpuUsageGaugeSet, it has done almost everything for you. You just need to make sure the com.alibaba.metrics.CachedMetricSet#refreshIfNecessary is called periodically so that you can always get the latest data.